### PR TITLE
Update to latest

### PR DESCRIPTION
**Pull request by pipo02mix**
Towards https://github.com/giantswarm/thg/issues/114

This PR:

- adds/changes/removes etc

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
               only: /^v.*/
+
+      - architect/run-tests-with-ats:
+          name: execute chart tests
+          app-test-suite_container_tag: "0.2.3"
+          filters:
+            # Do not trigger the job on merge to master.
+            branches:
+              ignore:
+                - master
```
Comment (ubergesundheit) [inline .circleci/config.yml]:
```suggestion
                - main
```

File: helm/linkerd2-multicluster-app/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+apiVersion: "v1"
+# this version will be updated by the CI before publishing the Helm tarball
+appVersion: edge-XX.X.X
```
Comment (ubergesundheit) [inline helm/linkerd2-multicluster-app/Chart.yaml]: IMO we should ideally sync to `stable` releases

File: helm/linkerd2-multicluster-app/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+description: |
+  The Linkerd-Multicluster extension contains resources to support multicluster
+  linking to remote clusters
+home: https://linkerd.io
+keywords:
+- service-mesh
+kubeVersion: ">=1.20.0-0"
+name: "linkerd-multicluster"
```
Comment (ubergesundheit) [inline helm/linkerd2-multicluster-app/Chart.yaml]: name should match the repository name without `-app` (See https://intranet.giantswarm.io/docs/dev-and-releng/app-developer-processes/adding_app_to_appcatalog/#11-create-and-configure-a-repo-for-the-app)

File: .ats/main.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+app-tests-skip-app-deploy: true
```
Comment (piontec) [inline .ats/main.yaml:1]: It seems all the tests are just copied over from linerd2-app repo. Should they be removed here?

File: helm/linkerd2-multicluster-app/README.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+![Version: 30.1.1-edge](https://img.shields.io/badge/Version-30.1.1--edge-informational?style=flat-square)
+
+![AppVersion: edge-XX.X.X](https://img.shields.io/badge/AppVersion-edge--XX.X.X-informational?style=flat-square)
+
+**Homepage:** <https://linkerd.io>
+
+## Quickstart and documentation
+
+You can run Linkerd on any Kubernetes 1.20+ cluster in a matter of seconds. See
+the [Linkerd Getting Started Guide][getting-started] for how.
```
Comment (piontec) [inline helm/linkerd2-multicluster-app/README.md]: Shouldn't we reference our linkerd2-app here?

File: kube.config [hunk 6]
```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
```
Comment (piontec) [inline kube.config]: What is this kube.config for? Should be included in the repo?

File: .ats/main.yaml (hunk 4 above)
Comment (pipo02mix) [reply .ats/main.yaml:1]: yeah but I added the multicluster deployment too and the linked ctl check refers to multicluster now

File: helm/linkerd2-multicluster-app/Chart.yaml (hunk 3 above)
Comment (pipo02mix) [inline helm/linkerd2-multicluster-app/Chart.yaml]:
```suggestion
name: "linkerd2-multicluster-app"
```

File: helm/linkerd2-multicluster-app/README.md (hunk 5 above)
Comment (pipo02mix) [reply helm/linkerd2-multicluster-app/README.md]: good point

Comment (pipo02mix) [inline helm/linkerd2-multicluster-app/README.md]:
```suggestion
the [Linkerd Getting Started Guide](https://github.com/giantswarm/linkerd2-app) for how.
```

**Comment by ubergesundheit** [comment]
@pipo02mix If you want, we could sync on bringing the app to our catalogs

**Comment by ubergesundheit** [comment]
In order to fix the latest CI failure, I think you need to also synchronize the `partials` helm chart, app-build-suite tries to verify/update the subchart but fails to do so. It does not help to have the subchart archive in the repository. I would suggest removing it.

File: .ats/main.yaml (hunk 4 above)
Comment (piontec) [reply .ats/main.yaml:1]: OK, but then it seems the tests are not executed here.

Comment (ubergesundheit) [reply .ats/main.yaml:1]: @pipo02mix  Lets keep them in here, we can use it as a starting point

File: kube.config (hunk 6 above)
Comment (pipo02mix) [reply kube.config]: removed

File: .abs/main.yaml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+chart-dir: helm/linkerd2-multicluster-target-app
+replace-chart-version-with-git: true
+replace-app-version-with-git: false
+generate-metadata: true
+catalog-base-url: https://giantswarm.github.io/giantswarm-catalog/
+# until we support custom template helper filenames
+disable-giantswarm-helm-validator: true
```
Comment (pipo02mix) [inline .abs/main.yaml]:
```suggestion
disable-giantswarm-helm-validator: true

```

File: .ats/main.yaml [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+
+smoke-tests-cluster-config-file: tests/kind_config.yaml
+smoke-tests-cluster-type: kind
+skip-steps: [functional, upgrade]
```
Comment (pipo02mix) [inline .ats/main.yaml]:
```suggestion
skip-steps: [functional, upgrade]

```

File: .circleci/config.yml [hunk 9]
```diff
@@ -1,17 +1,30 @@
...
+      - architect/run-tests-with-ats:
+          name: execute chart tests
+          app-test-suite_container_tag: "0.2.3"
+          filters:
+            # Do not trigger the job on merge to master.
+            branches:
+              ignore:
+                - main
+          requires:
+            - "package and push linkerd2-multicluster chart"
```
Comment (pipo02mix) [inline .circleci/config.yml]:
```suggestion
            - "package and push linkerd2-multicluster chart"

```

**Comment by pipo02mix** [comment]
Why this is still failing if I have included the partials chart?

https://circleci.com/gh/giantswarm/linkerd2-multicluster-target-app/9?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

@ubergesundheit ?

**Comment by ubergesundheit** [comment]
I'll try to resolve this on my machine. The partials need to be present in its yaml files form not as the charts tgz file

**Comment by ubergesundheit** [comment]
I added the partials from a stable tag, looks like upstream changed something in between. I'll re-initialize the main chart from the same stable tag (stable-2.11.2)

**Comment by ubergesundheit** [comment]
@pipo02mix Just talked to @webwurst about this

The plan is that @webwurst will prepare the mirror chart for the multicluster setup in a separate repository.

**Pull request merged**